pySDC/implementations/problem_classes/odeScalar.py

```diff
@@ -0,0 +1,213 @@
+
+    >>> class MyProtheroRobinson(ProtheroRobinson):
+    >>>
+    >>>     def g(self, t):
+    >>>         return np.exp(-0.2 * t)
+    >>>
+    >>>     def dg(self, t):
+    >>>         return (-0.2) * np.exp(-0.2 * t)
+
+    Parameters
+    ----------
+    epsilon : float, optional
+        Stiffness parameter. The default is 1e-3.
+    nonLinear : bool, optional
+        Wether or not to use the non-linear form of the problem. The default is False.
+    newton_maxiter : int, optional
+        Maximum number of Newton iteration in solve_system. The default is 200.
+    newton_tol : float, optional
+        Residuum tolerance for Newton iteration in solve_system. The default is 5e-11.
+    stop_at_nan : bool, optional
+        Wheter to stop or not solve_system when getting NAN. The default is True.
+
+    Reference
+    ---------
+    A. Prothero and A. Robinson, On the stability and accuracy of one-step methods for solving
+    stiff systems of ordinary differential equations, Mathematics of Computation, 28 (1974),
+    pp. 145–162.
+    """
+
+    dtype_u = mesh
+    dtype_f = mesh
+
+    def __init__(self, epsilon=1e-3, nonLinear=False, newton_maxiter=200, newton_tol=5e-11, stop_at_nan=True):
+        nvars = 1
+        super().__init__((nvars, None, np.dtype('float64')))
+
+        self.f = self.f_NONLIN if nonLinear else self.f_LIN
+        self.jac = self.jac_NONLIN if nonLinear else self.jac_LIN
+        self._makeAttributeAndRegister(
+            'epsilon', 'nonLinear', 'newton_maxiter', 'newton_tol', 'stop_at_nan', localVars=locals(), readOnly=True
+        )
+        self.work_counters['newton'] = WorkCounter()
+        self.work_counters['rhs'] = WorkCounter()
+
+    # -------------------------------------------------------------------------
+    # g function (analytical solution), and its first derivative
+    # -------------------------------------------------------------------------
+    def g(self, t):
+        return np.cos(t)
+
+    def dg(self, t):
+        return -np.sin(t)
+
+    # -------------------------------------------------------------------------
+    # f(u,t) and Jacobian functions
+    # -------------------------------------------------------------------------
+    def f(self, u, t):
+        raise NotImplementedError()
+
+    def f_LIN(self, u, t):
+        return -self.epsilon ** (-1) * (u - self.g(t)) + self.dg(t)
+
+    def f_NONLIN(self, u, t):
+        return -self.epsilon ** (-1) * (u**3 - self.g(t) ** 3) + self.dg(t)
+
+    def jac(self, u, t):
+        raise NotImplementedError()
+
+    def jac_LIN(self, u, t):
+        return -self.epsilon ** (-1)
+
+    def jac_NONLIN(self, u, t):
+        return -self.epsilon ** (-1) * 3 * u**2
+
+    # -------------------------------------------------------------------------
+    # pySDC required methods
+    # -------------------------------------------------------------------------
+    def u_exact(self, t, u_init=None, t_init=None):
```

Comment:
Same thing with `u_init` and `t_init`. Maybe there is a VS code functionality that lets you do all these at once? :P